src/ecalc/libraries/libecalc/common/libecalc/core/consumers/consumer_system.py

```diff
@@ -180,29 +191,44 @@ def evaluate(
                         regularity=temporal_operational_settings.models[0].model[0].rates[0].regularity,
                     ),
                 )
+
+                energy_usage[valid_indices] = energy_usage_for_period[valid_indices_for_period].values
                 power[valid_indices] = power_for_period[valid_indices_for_period].values
+                operational_settings_used[valid_indices] = [operational_setting_index + 1 for _ in valid_indices]
+
+                # Storing all consumer results for all calculated operational settings.
+                all_consumer_results[str(period)][operational_setting_index] = consumer_results
 
-                operational_settings_used[valid_indices] = operational_setting_index + 1  # 1-based index
+                # Fixme: Unit data should come from the consumer itself, not the results.
+                energy_usage.unit = consumer_results[0].component_result.energy_usage.unit
+                power.unit = consumer_results[0].component_result.power.unit
 
                 if reduce(operator.mul, is_valid.values):
                     # quit as soon as all time-steps are valid. This means that we do not need to test all settings.
                     break
+                elif operational_setting_index + 1 == len(operational_settings):
+                    # If we are at the last operational_setting and not all indices are valid
+                    invalid_indices = [i for i, x in enumerate(is_valid.values) if not x]
+                    energy_usage[invalid_indices] = list(np.nan_to_num(energy_usage_for_period[invalid_indices].values))
+                    power[invalid_indices] = list(np.nan_to_num(power_for_period[invalid_indices].values))
+                    operational_settings_used[invalid_indices] = [
+                        operational_setting_index + 1 for _ in invalid_indices
+                    ]
 
         consumer_result = ConsumerSystemResult(
             id=self.id,
             is_valid=is_valid,
             timesteps=variables_map.time_vector,
-            energy_usage=energy_usage.to_calendar_day(),
+            energy_usage=energy_usage.to_stream_day()
```

Comment:
I asked, dont think I got answer. reason to change to stream_day here is to support consumer systems in consumer systems? It means that the end values will be in stream day as well? (I think that is ok, as long as we specify it, like we do), but some places may need and update for this? (not sure)

Reply:
This is done in order to align consumer system v1 and v2.